src/SquidMath.h

```diff
@@ -123,5 +123,40 @@ SetToNaturalSumOrMax(S &var, Args... args)
     return var;
 }
 
+template <typename P, typename T, typename U>
+Optional<P>
+IntegralProduct(P, T t, U u)
```

Comment:
Since we treat negative numbers specially, this should be called NaturalProduct(), just like the sum function that uses the same special treatment of negative numbers. IIRC, natural implies integral, so we are not loosing any meaning by renaming. This name will affect a few related APIs/names like `testIntegralProduct`.